perception/navigator_vision/nodes/cvod/cvod.py

```diff
@@ -0,0 +1,263 @@
+        except CvBridgeError as e:
+            print(e)
+
+        # Figure out what camera we are looking at.
+        if(camera == "seecam"):
+        	# Tick up the counter.
+            self.see_frame_counter = self.see_frame_counter + 1
+            if(self.see_frame_counter >= 2):
+	        	# After a minimum number of frames, update PCODAR. This should occur
+	        	# roughly one time every second.
+	        	self.update_pcodar(boxes, classes, scores, camera)
+	        	self.see_frame_counter = 0
+
+	    # We are looking at the left cam.
+        else:
+        	# Tick up counter.
+        	self.left_frame_counter = self.left_frame_counter + 1
+        	if(self.left_frame_counter >= 2):
+	        	# After a minimum number of frames, update PCODAR. This should occur
+	        	# roughly one time every second.
+	        	self.update_pcodar(boxes, classes, scores, camera)
+	        	self.left_frame_counter = 0
+
+        return (boxes, classes)
+
+    def update_pcodar(self, boxes, classes, scores, camera_name):
+    	# Determine exactly where the image came from and retrieve the necessary
+    	# information.
+    	if camera_name == "/seecam":
+    		camera = self.see_sub
+    		img_geom = self.see_img_geom
+        else:
+    		camera = self.left_sub
+    		img_geom = self.left_img_geom
+
+    	# Get the transformation to allow us to get usable info from camera
+    	# pixels.
+        time = camera.last_image_time
+        self.tf_listener.waitForTransform("/enu", camera_name, rospy.Time(0), rospy.Duration(10))
+
+        enu_cam_tf = self.tf_listener.lookupTransform("/enu", camera_name, rospy.Time(0))
+
+        if not enu_cam_tf:
+            rospy.logwarn("No image recieved yet, are you sure the camera is publishing the right topic?")
+            return
+        cam_p, cam_q = enu_cam_tf
+        cam_p = np.array(cam_p)
+        cam_r = tf.transformations.quaternion_matrix(cam_q)[:3, :3]
+        # Fetch the current objects on the Ogrid.
+        all_objects = None
+        try:
+            all_objects = self.DBQuery_proxy(name="all", cmd="").objects
+        except rospy.ServiceException, e:
+            print ("Couldn't call database")
+            rospy.sleep(5)
+            return
+        if all_objects is None:
+            print ("Didn't find any objects")
+            rospy.sleep(5)
+            return
+        # Attempt to label the objects we have found.
+        points, labels = boxes, classes
+        for j in range(len(points)):
+            if scores[0][j] < 0.80:
+                continue 
+            for k in range(len(points[j])):
+                ray = self.get_ray_in_enu(points[j][k], img_geom, cam_r)
+            if ray is None:
+                continue
+            o = self.find_closest_object_given_ray(all_objects, ray, cam_p)
+            if o is not None:
+                self.set_label_for_object(labels[0][j], o)
+            rospy.sleep(2)
```

Comment:
There are several of these sleeps throughout the program that I'm not sure are useful/needed.